test/tests/userobjects/setup_interface_count/tests

```diff
@@ -0,0 +1,60 @@
+# The following tests are designed to count how often the setup methods are called (initialSetup, etc.),
+# which requires that the threading model and number of threads be set to known values.
+#
+# Also, GeneralUserObject must also be restricted for a certain MPI because there methods will be called
+# on each processor, since they are not distributed in any fashion (e.g., on elements)
+
+[Tests]
+  [./GeneralUserObject]
+    type = CSVDiff
+    input = general.i
+    csvdiff = general_out.csv
+    min_threads = 2
+    max_threads = 2
+    min_parallel = 2
+    max_parallel = 2
+    tbb = false
+    openmp = false
+    pthreads = true
```

Comment:
The best solution is probably to add new flags to libMesh, USING_TBB, etc.
and we can key off of these.

On Mon, Feb 29, 2016 at 3:15 PM, Cody Permann <notifications@github.com>
wrote:

> In test/tests/userobjects/setup_interface_count/tests
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_idaholab_moose_pull_6375-23discussion-5Fr54488269&d=BQMCaQ&c=54IZrppPQZKX9mLzcGdPfFD1hxrcB__aEkJFOKJFd00&r=h7heP8xwI1i_HikChvhFbEBurKirgfOCdwgBxB9lM8c&m=Cy_DhPFvtSZeRIZntwHnv2KA9ddrjQUkKRNu1_OTudM&s=ZPnaeRRG8FxjiaUmANqdo_nRNX-5YJMyk3q_OTx1pHo&e=>
> :
>
> > +#
> > +# Also, GeneralUserObject must also be restricted for a certain MPI because there methods will be called
> > +# on each processor, since they are not distributed in any fashion (e.g., on elements)
> > +
> > +[Tests]
> > +  [./GeneralUserObject]
> > +    type = CSVDiff
> > +    input = general.i
> > +    csvdiff = general_out.csv
> > +    min_threads = 2
> > +    max_threads = 2
> > +    min_parallel = 2
> > +    max_parallel = 2
> > +    tbb = false
> > +    openmp = false
> > +    pthreads = true
>
> Actually that's definitely not true:
> libMesh normally picks up pthreads all the time on our boxes but it has a
> lower precedent than tbb. Furthermore, you can actually turn on all three
> models and it'll run with tbb. You don't have to explicitly turn off the
> models with lower precedence. See why I don't like these flags? [image:
> :smile:]
>
> —
> Reply to this email directly or view it on GitHub
> <https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_idaholab_moose_pull_6375_files-23r54488269&d=BQMCaQ&c=54IZrppPQZKX9mLzcGdPfFD1hxrcB__aEkJFOKJFd00&r=h7heP8xwI1i_HikChvhFbEBurKirgfOCdwgBxB9lM8c&m=Cy_DhPFvtSZeRIZntwHnv2KA9ddrjQUkKRNu1_OTudM&s=yGiDC6oPe-YQfRPmp0AJFA_a1LP4whfhyQBDs181ffg&e=>
> .
>